doc/src/arch/reference.rst

```diff
@@ -1897,6 +1902,98 @@ The clocking configuration is specified with ``<clock>`` tags within the ``<cloc
+
+.. arch:tag:: <clock_network name="string" num_inst="integer">
+
+    :req_param name: A unique string for reference.
+    :req_param num_inst: which describes the number of parallel instances of the clock distribution types described in the ``<clock_network>`` sub-elements.
+
+The supported clock distribution types are ``<spine>`` and ``<rib>``. ``Spine`` is used to describe vertical clock distribution wires. Whereas, ``Rib`` is used to describe a horizontal clock distribution wire.
+
+.. arch:tag:: <spine metal_layer="string" x="integer" starty="integer" endy="integer" repeatx="integer" repeaty="integer"/>
+    :req_param metal_layer: A referenced metal layer that sets the unit resistance and capacitance of the distribution wire over the length of the wire.
+    :req_param starty: The start y grid location, of the wire which runs parallel to the y-axis from starty and ends at endy, inclusive. Value can be relative to the device size.
+    :req_param endy: The end of y grid location of the wire. Value can be relative to the device size.
+    :req_param x: The location of the spine with respect to the x-axis. Value can be relative to the device size.
+    :opt_param repeatx: The horizontal repeat factor of the spine along the device. Value can be relative to the device size.
+    :opt_param repeaty: The vertical repeat factor of the spine along the device. Value can be relative to the device size.
+
+The provided example clock network defines two spines, and neither repeats as each spans the entire height of the device and is locally at the horizontal midpoint of the device.
+
+.. arch:tag:: <rib metal_layer="string" y="integer" startx="integer" endx="integer" repeatx="integer" repeaty="integer"/>
+    :req_param metal_layer: A referenced metal layer that sets the unit resistance and capacitance of the distribution wire over the length of the wire.
+    :req_param startx: The start x grid location, of the wire which runs parallel to the x-axis from startx and ends at endx, inclusive. Value can be relative to the device size.
+    :req_param endx: The end of x grid location of the wire. Value can be relative to the device size.
+    :req_param y: The location of the rib with respect to the y-axis. Value can be relative to the device size.
+    :opt_param repeatx: The horizontal repeat factor of the rib along the device. Value can be relative to the device size.
+    :opt_param repeaty: The vertical repeat factor of the rib along the device. Value can be relative to the device size.
+
+Along each spine and rib is a group of switch points. Switch points are used to describe drive or tap locations along the clock distribution wire:
+
+.. arch:tag:: <switch_point type="{drive | tap}" name="string" yoffset="integer" xoffset="integer" xinc="integer" yinc="integer" buffer="string">
```

Comment:
I think we'll need diagrams to illustrate this, other wise it's likely unclear what a switchpoint is.

If the earlier rib-spine diagrams aren't too busy we could annotate the switchpoints on them. However I expect it may be better to create new figures using the same illustrative base diagrams with switchpoint annotations instead.